cpp/include/cudf/lists/detail/scatter.cuh

```diff
@@ -45,105 +37,6 @@ namespace detail {
 
 namespace {
```

Comment:
Can this be removed? Anonymous namespaces should not be in headers.

Reply:
Yup, I think everything inside can also be moved to `scatter_helper`